packages/snaps-controllers/src/snaps/SnapController.test.tsx

```diff
@@ -10231,19 +10239,39 @@ describe('SnapController', () => {
 
       expect(snapController.has(MOCK_SNAP_ID)).toBe(true);
 
+      const requestPromise = snapController.handleRequest({
+        snapId: MOCK_SNAP_ID,
+        origin: METAMASK_ORIGIN,
+        handler: HandlerType.OnRpcRequest,
+        request: {
+          method: 'foo',
+        },
+      });
+
+      await waitForStateChange(messenger);
+
+      expect(snapController.isRunning(MOCK_SNAP_ID)).toBe(true);
+
+      await new Promise((resolve) => setTimeout(resolve, 100));
+
       await snapController.clearState();
 
       expect(snapController.has(MOCK_SNAP_ID)).toBe(false);
 
       expect(callActionSpy).toHaveBeenCalledWith(
-        'ExecutionService:terminateAllSnaps',
+        'ExecutionService:terminateSnap',
+        MOCK_SNAP_ID,
       );
 
       expect(callActionSpy).toHaveBeenCalledWith(
         'PermissionController:revokeAllPermissions',
         MOCK_SNAP_ID,
       );
 
+      await expect(requestPromise).rejects.toThrow(
+        'npm:@metamask/example-snap was stopped and the request was cancelled. This is likely because the Snap crashed.',
+      );
+
```

Comment:
Doesn't this means that if the snap is pending a request when we reset the state this error will be thrown?

Reply:
I'm afraid that we will see some Sentry errors rising in the controllers that doesn't catch the error